### PR TITLE
goreleaser: Some adjustments

### PR DESCRIPTION
**Pull request by tuxillo**
- Rework the Tag detection a bit
- Exit if build fails
- Use 'openssl md5' instead of md5sum which is not available in
  some non-Linux systems.

### CODE REVIEW

File: goreleaser.sh [hunk 1]
```diff
@@ -16,14 +16,17 @@
...
 
-Tag=$(git tag --points-at HEAD | grep v[0-9.0-9.0-9])
-FullCommit=$(git rev-parse HEAD)
+# Tries to identify a tag name and defaults to branch name if it fails
+Tag=$(git tag --points-at HEAD | grep "v[0-9]\.[0-9]\.[0-9]*")
+Tag=${Tag:-$(git rev-parse --abbrev-ref HEAD)}
 
-if [[ $(uname) == "Darwin" ]]
+# Assumes anything non-linux has date -u
+Osname=$(uname)
```
Comment (iknite) [inline goreleaser.sh]: Why do it need the extra variable?

**Review by suizman — APPROVED** [review]
Thank you for your contribution. We're glad to hear that other people use QED in distinct environments.
LGTM

File: goreleaser.sh (hunk 1 above)
Comment (suizman) [reply goreleaser.sh]: I like the idea. That enables you to do further checks.

**Comment by suizman** [comment]
Merging

**Comment by iknite** [comment]
I believe there's a regression in master branch to `v0.1` 

the output of `git fetch -ap && git show-tree`:
```
* ff51d824 (master, origin/master) Merge pull request #181 from tuxillo/goreleaser-1
| * f96dd806 (tag: v1.0.0-rc1) Fix Azure release pattern
| * 3c3a3b05 (origin/azure-pipelines-release) Update Azure Pipelines and release script
| * e3223991 (tag: v0.3.0-docs) Merge pull request #146 from jllucas/restore
| * e90d4b33 (tag: v0.2.4-docs) Fix: panic when version is not set in cmd.
| | * 1523582f (origin/dragonboat) WIP
| |/
| | * cf9514b1 (origin/demo) doc: Update README
| |/
| * b3e096e9 (tag: v0.2-alpha) Fix goreleaser environment variables
| | * 8b735da8 (origin/tampering) Update scripts to support MacOS execution
| | * 06127d8f (tag: tampering) update intro documentation
| |/
| * c4ea9d35 (tag: v0.2-M3) Update archive files
| * c44c8b42 (tag: v0.2-M2) Use incremental ports in tests to prevent address already in use
| * c1061353 (tag: v0.2-M1) Remove Mutexes and rename EdSigner to Ed25519Signer
|/
* 32380634 (tag: v0.1) README.md: Update instructions to follow current cli interface, misc fixes
* 4b8cf79a initial push
```
I believe it needs to be `--rebased` before merged to master

**Comment by iknite** [comment]
Something weird is happening to this command diff-ing between commits is *OK* maybe it's something @panchoh will like to dig into, but it's nothing to be worried. Thanks, guys

**Comment by suizman** [comment]
> Something weird is happening to this command diff-ing between commits is _OK_ maybe it's something @panchoh will like to dig into, but it's nothing to be worried. Thanks, guys

Some time ago we've rewrite the repo history in order to remove binary files. It's possible this causes the inconsistency.
Please open an separate issue to explore this situation.

**Pull request merged**